### PR TITLE
AArch64 projects: add 32-bit userspace support

### PR DESCRIPTION
**Pull request by Raybuntu**
OK here I go trying to upstream this hacky monster. 
I'm introducing a possibility to build aarch64 projects with 32bit userland. 
All kernel packages are build using a static aarch64 linaro toolchain. Me and @kszaq are using "it" for almost a year now in our community builds. @kszaq, @Kwiboo  and me have been improving the code and we upstreamed all but one missing compat ioctl patches to: https://github.com/LibreELEC/linux-amlogic

To make DVB work with 32bit userspace we need one additional patch and I've already created a PR for that:
https://github.com/LibreELEC/linux-amlogic/pull/61

I've tested a clean Odroid_C2.arm build with NF and DAZN.

### CODE REVIEW

File: packages/emulation/libretro-beetle-pcfx/package.mk [hunk 1]
```diff
@@ -51,11 +51,12 @@ make_target() {
     imx6)
       make platform=armv7-cortexa9-neon-hardfloat
       ;;
-    WeTek_Play|WeTek_Core)
-      make platform=armv7-cortexa9-neon-hardfloat
-      ;;
-    Odroid_C2|WeTek_Hub|WeTek_Play_2)
-      make platform=aarch64
+    WeTek_Play|WeTek_Core|Odroid_C2|WeTek_Hub|WeTek_Play_2)
+      if [ $TARGET_ARCH = 'aarch64' ];then
```
Comment (MilhouseVH) [inline packages/emulation/libretro-beetle-pcfx/package.mk]: `[ "$TARGET_ARCH" = "aarch64" ]` for consistency?

File: packages/graphics/opengl-meson/package.mk [hunk 2]
```diff
@@ -29,8 +29,12 @@ case $MESON_FAMILY in
     PKG_SHA256="21a8376668c84bf1b9e64a917fcfa1cf74689035fed8e4630833c9cde28d40c1"
     ;;
   gxbb)
-    PKG_VERSION="gxbb-r5p1-01rel0"
-    ;;
+    if  [ "$TARGET_ARCH" = "arm"  ];then
```
Comment (MilhouseVH) [inline packages/graphics/opengl-meson/package.mk]: Extra space after `"arm"` can be removed.

Comment (Raybuntu) [reply packages/graphics/opengl-meson/package.mk]: Thanks. I will fix that.

File: packages/emulation/libretro-beetle-pcfx/package.mk (hunk 1 above)
Comment (Raybuntu) [reply packages/emulation/libretro-beetle-pcfx/package.mk]: Ahhh I see. Thanks I'm gonna fix it.

File: projects/WeTek_Play_2/options [hunk 3]
```diff
@@ -21,6 +21,14 @@
         TARGET_CPU="cortex-a53"
         TARGET_CPU_FLAGS="+crc+fp+simd"
         ;;
+      arm)
+	TARGET_KERNEL_ARCH="arm64"
```
Comment (chewitt) [inline projects/WeTek_Play_2/options]: it's a nitpick, but indentation is off here :)

**Comment by Raybuntu** [comment]
Thanks @lrusak.
@chewitt Should be fixed now. Stupid "tab/spaces" mixup. I don't see it cause my vim fixes it for me visually.

File: packages/emulation/libretro-beetle-pcfx/package.mk [hunk 4]
```diff
@@ -51,11 +51,12 @@ make_target() {
     imx6)
       make platform=armv7-cortexa9-neon-hardfloat
       ;;
-    WeTek_Play|WeTek_Core)
-      make platform=armv7-cortexa9-neon-hardfloat
-      ;;
-    Odroid_C2|WeTek_Hub|WeTek_Play_2)
-      make platform=aarch64
+    WeTek_Play|WeTek_Core|Odroid_C2|WeTek_Hub|WeTek_Play_2)
+      if [ $TARGET_ARCH = "aarch64" ];then
```
Comment (Kwiboo) [inline packages/emulation/libretro-beetle-pcfx/package.mk]: nitpick: quote around `$TARGET_ARCH` to be consistent
nitpick: add space before `then` to be consistent (same goes for the rest of packages)

File: packages/graphics/opengl-meson/package.mk [hunk 5]
```diff
@@ -29,8 +29,12 @@ case $MESON_FAMILY in
     PKG_SHA256="21a8376668c84bf1b9e64a917fcfa1cf74689035fed8e4630833c9cde28d40c1"
     ;;
   gxbb)
-    PKG_VERSION="gxbb-r5p1-01rel0"
-    ;;
+    if  [ "$TARGET_ARCH" = "arm" ];then
```
Comment (Kwiboo) [inline packages/graphics/opengl-meson/package.mk]: nitpick: double space in `if  [`
nitpick: add space before `then` to be consistent

File: scripts/image [hunk 6]
```diff
@@ -222,6 +222,9 @@ find $INSTALL/usr/lib/modules/$MODVER/ -name *.ko | \
 $TOOLCHAIN/bin/depmod -b $INSTALL/usr -a -e -F "$BUILD/linux-$(kernel_version)/System.map" $MODVER
 
 # strip kernel modules
+if [ "$TARGET_KERNEL_ARCH" = "arm64" -a "$TARGET_ARCH" = "arm"  ]; then
```
Comment (Kwiboo) [inline scripts/image]: nitpick: double space in `"  ];`

**Comment by Raybuntu** [comment]
@Kwiboo Thanks. I fixed those and a couple more of those missing spaces before "then".

**Comment by chewitt** [comment]
Thanks @Raybuntu @kszaq @Kwiboo for all the testing and review 🥇

**Pull request merged**